src/project-generator.ts

```diff
@@ -51,266 +39,97 @@ export class ProjectGenerator {
-                    tree: [
-                        {
-                            path: 'app/AppManifest.json',
-                            mode: GIT_DATA_MODES.fileBlob,
-                            type: GIT_DATA_TYPES.blob,
-                            sha: blobSha1,
-                        },
-                        {
-                            path: 'app/src/main.py',
-                            mode: GIT_DATA_MODES.fileBlob,
-                            type: GIT_DATA_TYPES.blob,
-                            sha: blobSha2,
-                        },
-                    ],
-                    base_tree: baseTreeSha,
-                },
-                this.requestConfig
-            );
-            const newTreeSha = response.data.sha;
-            return newTreeSha;
-        } catch (error) {
-            throw error;
-        }
+    private async getNewMainPySha(appName: string, codeSnippet: string): Promise<string> {
+        const mainPyContentData = await this.gitRequestHandler.getFileContentData('main');
+        const decodedMainPyContentData = decode(mainPyContentData);
+        const decodedBase64CodeSnippet = decode(codeSnippet);
+        const convertedMainPy = this.codeConverter.convertMainPy(decodedMainPyContentData, decodedBase64CodeSnippet, appName);
+        const encodedConvertedMainPy = encode(`${convertedMainPy}\n`);
+        const mainPyBlobSha = await this.gitRequestHandler.createBlob(encodedConvertedMainPy);
+        return mainPyBlobSha;
     }
 
-    private async getMainBranchSha(): Promise<string> {
-        try {
-            const response = await axios.get(`${this.repositoryPath}/git/refs/heads/main`, this.requestConfig);
-            const mainBranchSha = response.data.object.sha;
-            return mainBranchSha;
-        } catch (error) {
-            throw error;
-        }
-    }
-
-    private async createCommitSha(mainBranchSha: string, newTreeSha: string): Promise<string> {
-        try {
-            const response = await axios.post(
-                `${this.repositoryPath}/git/commits`,
-                {
-                    tree: newTreeSha,
-                    message: DEFAULT_COMMIT_MESSAGE,
-                    parents: [mainBranchSha],
-                },
-                this.requestConfig
-            );
-            const commitSha = response.data.sha;
-            return commitSha;
-        } catch (error) {
-            throw error;
-        }
-    }
-
-    private async updateMainBranchSha(newCommitSha: string): Promise<any> {
-        try {
-            const response = await axios.patch(
-                `${this.repositoryPath}/git/refs/heads/main`,
-                {
-                    sha: newCommitSha,
-                },
-                this.requestConfig
-            );
-            const patchResponse = response.data;
-            return patchResponse;
-        } catch (error) {
-            throw error;
+    private vspecIsPayload(vspec: string): boolean {
```

Comment:
I recommend we get rid of this by introducing the changes I have suggested above.